src/test/java/co/rsk/federate/watcher/FederationWatcherTest.java

```diff
@@ -79,248 +76,317 @@ void whenFederationWatcherIsSetUp_shouldAddListener() throws Exception {
     }
 
     @Test
-    void whenNoActiveFederation_shouldTriggerActiveFederationChange() throws Exception {
+    void onBestBlock_whenNoActiveFederationAndProposedFederationExists_shouldTriggerActiveAndProposedFederationChange() throws Exception {
         // Arrange
-        var rskListener = setupAndGetRskListener(null, null);
-        var activeCalls = new AtomicInteger(0);
-        var retiringCalls = new AtomicInteger(0);
-
+        var rskListener = setupAndGetRskListener(null, null, null);
+        when(federationProvider.getProposedFederationAddress()).thenReturn(Optional.of(SECOND_FEDERATION.getAddress()));
+        when(federationProvider.getProposedFederation()).thenReturn(Optional.of(SECOND_FEDERATION));
         when(federationProvider.getActiveFederationAddress()).thenReturn(FIRST_FEDERATION.getAddress());
         when(federationProvider.getActiveFederation()).thenReturn(FIRST_FEDERATION);
         when(federationProvider.getRetiringFederationAddress()).thenReturn(Optional.empty());
```

Comment:
nit, but: i'd put the proposed federation below the retiring, to match the order we have in the rskListener